<launch>
  <node name="image_sim_node" pkg="yak_ros" type="yak_ros_image_simulator">
    <param name="base_frame" value="world"/>
    <param name="mesh" value="$(find yak_ros)/demo/bun_on_table.ply"/>
    <param name="orbit_speed" value="1.0"/>
    <param name="framerate" value="30"/>
  </node>


  <node name="tsdf_node" pkg="yak_ros" type="yak_ros_node">
    <remap from="~input_depth_image/image" to="/sim_depth_camera/image"/>
    <remap from="~input_depth_image/camera_info" to="/sim_depth_camera/camera_info"/>

    <param name="tsdf_frame" value="tsdf_origin"/>

    <param name="volume_resolution" value="0.001"/>
    <param name="volume_x" value="640"/>
    <param name="volume_y" value="640"/>
    <param name="volume_z" value="192"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="-0.3 -0.3 -0.01 0 0 0 1 world tsdf_origin 100" />
</launch>
